### ISSUE TITLE
LargePayloadTest fails with SwiftNIO 1.7.0

### ISSUE THREAD

**Issue by pushkarnk** [IBM-Swift/Kitura-NIO#7]
`LargePayloadTests.testLargeGets` fails consistently with SwiftNIO 1.7.0 with:

```
*** Error in `/root/pushkar/Kitura-NIO/.build/x86_64-unknown-linux/debug/KituraNIOPackageTests.xctest': realloc(): invalid next size: 0x00007f5e4004c110 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7f5ea5d447e5]
/lib/x86_64-linux-gnu/libc.so.6(+0x834aa)[0x7f5ea5d504aa]
/lib/x86_64-linux-gnu/libc.so.6(realloc+0x179)[0x7f5ea5d51839]
```

**Comment by pushkarnk** [comment]
This is an isolated recreate for the bug:

```
import KituraNIO
import Foundation

class TestServerDelegate : ServerDelegate {
    func handle(request: ServerRequest, response: ServerResponse) {
        handleGet(request: request, response: response)
    }

    func handleGet(request: ServerRequest, response: ServerResponse) {
        let payloadData = Data(repeating: 65, count: 10658240)
        do {
            response.headers["Content-Length"] = ["\(payloadData.count)"]
            try response.write(from: payloadData)
            try response.end()
        }
        catch {
            print("Error writing response.")
        }
    }
}

let httpServer = HTTPServer()
httpServer.delegate = TestServerDelegate()
try! httpServer.listen(on: 0)

let request = HTTP.request("http://127.0.0.1:\(httpServer.port!)/") { response in
    print(response!.statusCode)
}
request.end()
Dispatch.dispatchMain()
```

This crash happens very frequently after swift-no 1.7.0. It was extremely intermittent with 1.6.1. 
Reported this issue to the Swift-NIO team via Slack!

**Comment by pushkarnk** [comment]
Fixed in swift-nio 1.8
